pymodbus/client/serial.py

```diff
@@ -175,7 +177,7 @@ def __init__(
             stopbits=stopbits,
             **kwargs,
         )
-        self.socket = None
+        self.socket: serial.Serial | None = None
```

Comment:
I have to check this one, I am pretty sure it a leftover from when transport was integrated.

Reply:
Ahh it in the sync code, that is correct.

Reply:
Just FYI, improving this type hint is what enabled the type checkers to find https://github.com/pymodbus-dev/pymodbus/pull/2031